tinkergraph-gremlin/src/main/java/org/apache/tinkerpop/gremlin/tinkergraph/structure/AbstractTinkerGraph.java

```diff
@@ -563,6 +563,34 @@ public Object convert(final Object id) {
             public boolean allow(final Object id) {
                 return true;
             }
+        },
+        
+        /**
+         * Manages identifiers of type {@code String}.
+         */
+        STRING {
+            @Override
+            public String getNextId(final AbstractTinkerGraph graph) {
+                return java.util.UUID.randomUUID().toString();
+            }
+
+            @Override
+            public Object convert(final Object id) {
+                if (null == id)
+                    return null;
+                else  if (id instanceof String) {
+                    if (((String)id).isEmpty())
+                        throw new IllegalArgumentException("Expected a non-empty string but received an empty string.");
+
+                    return id;
+                } else
+                    throw new IllegalArgumentException(createErrorMessage(java.lang.String.class, id));
+            }
+
+            @Override
+            public boolean allow(final Object id) {
+                return id instanceof String && !((String)id).isEmpty();
```

Comment:
A quick thought; this seems a bit restrictive since the default returns a UUID string. Maybe this should be extended to support more types? All objects (as in `String.valueOf(Object)`) might be too broad but maybe at least UUID?

Reply:
I thought about that, but if there's string ids "xxxx-yyyy-...." and "xxxxyyy..." in a database, and your're accepting UUID {xxxx-yyyy....}, which one is it?

(The form is obviously a bit off, it's about the hyphens, you get the idea)

Reply:
Of course, if we define the canonical string representation to be just the result of toString, there's no ambiguity. I can revise the PR if agreed upon.

Reply:
> I thought about that, but if there's string ids "xxxx-yyyy-...." and "xxxxyyy..." in a database, and your're accepting UUID {xxxx-yyyy....}, which one is it?

That's a good point, but I'm not sure it fully applies here. In embedded mode, you're likely already using Java, and in remote mode, it would be deserialized into a Java UUID. So in the end, Java's string representation of UUID is effectively the correct form.

Anyways, I'm fine with this PR (already voted) without needing this change. Making it less restrictive is something that can easily be done in the future.